### PR TITLE
emocli: init at 1.0.0

### PR DESCRIPTION
**Pull request by AsbjornOlling**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MikaelFangel** [comment]
Your PR should contain two commits:
```
maintainers: add <handle>
emocli: init at 1.0.0
```

File: pkgs/by-name/em/emocli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/em/emocli/package.nix:9]: As this looks like it is your repo, I think it would be best that you made a tag for your release and to replace this line with:
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/em/emocli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/em/emocli/package.nix:29]:
```suggestion
    maintainers = with maintainers; [ asbjornolling ];
    mainProgram = "emocli";
```

File: pkgs/by-name/em/emocli/package.nix
Comment (AsbjornOlling) [reply pkgs/by-name/em/emocli/package.nix]: Good idea! Looks like I even used git tags on the last release :)
Fixed.

File: pkgs/by-name/em/emocli/package.nix
Comment (AsbjornOlling) [reply pkgs/by-name/em/emocli/package.nix:29]: Looks like I missed that tag. Thanks.
Fixed :)

**Comment by AsbjornOlling** [comment]
Oops.. accidentally had my maintainer githubId in quotes, which was what failed ofborg. Should be fixed now.

File: pkgs/by-name/em/emocli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/em/emocli/package.nix:9]: Because of how you created the tag this should be:
```suggestion
    rev = version;
```

File: pkgs/by-name/em/emocli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/em/emocli/package.nix:22]: Also, I think you might have a licensing issue, as I can't see to find the below license anywhere in your project?

[https://www.unicode.org/license.txt](https://www.unicode.org/license.txt)

I suggest that this gets added to the nix package.

Comment (AsbjornOlling) [reply pkgs/by-name/em/emocli/package.nix:22]: This has left me a bit confused, since emocli doesn't ship with the unicode data files.
It reads the data files at compile-time, in order to obtain all the emoji symbols.

The resulting binary *does* include unicode characters, but surely it's not required to include the unicode license whenever I use a unicode character?
If that's the case, it becomes a hassle to spell my name, or the city I live in.

:man_shrugging: but including the license shouldn't be a problem anyway, regardless of whether it's necessary or not.

I couldn't find a way to add multiple licenses to the `meta` section of the emocli package, so I figured that I'd have to make the unicode data files a separate package.
Luckily, there already is the `unicode-emoji` package in nixpkgs! That package correctly lists the unicode license in the `meta` section. Using the existing package is cleaner anyways.

There is the small issue of nixpkg's `unicode-emoji` being at an older version, so it's missing a few emoji compared to before. But I'll just make a separate PR to update the `unicode-emoji` package, once this one is merged :slightly_smiling_face:

Comment (AsbjornOlling) [reply pkgs/by-name/em/emocli/package.nix]: I've pushed a new verison of the package, that now reads the unicode data files from the `unicode-emoji` nix packge, instead of getting them using fetchurl. Hopefully that resolves any license problems.

Comment (MikaelFangel) [reply pkgs/by-name/em/emocli/package.nix]: You're correct about you don't have to include the license whenever you use a Unicode character. Without being an expert on licenses. Then as I read it the problem is that you distribute software containing the Unicode emojis but if there already is a package for the emoji's it would be better to use that and let that package handle the licensing. Sorry for not noticing it before 😊

**Comment by MikaelFangel** [comment]
Result of nixpkgs-review:
<details>
<summary>1 package built:</summary>
emocli
</details>

**Review by MikaelFangel — APPROVED** [review]

File: pkgs/by-name/em/emocli/package.nix
Comment (AsbjornOlling) [reply pkgs/by-name/em/emocli/package.nix]: @MikaelFangel 
No worries at all. I wouldn't expect you to read the source code of my project :)

Thanks for being patient with me, and thanks for reviewing my nix code!

**Comment by AsbjornOlling** [comment]
I changed a few things:
- rely on libc from the stdenv instead of explicitly statically linking musl.
- write `env.EMOCLI_DATAFILE` instead, for compatibility with structuredAttrs

**Review by lilyinstarlight — APPROVED** [review]
Looks good to me. Thank you and welcome to nixpkgs!

**Pull request merged**
